lib/spack/spack/test/config.py

```diff
@@ -1392,3 +1392,45 @@ def highest_priority_scopes(config, *, nscopes):
         )
 
     assert highest_priority_scopes(spack.config.CONFIG, nscopes=2) == expected_scopes_without_env
+
+
+@pytest.mark.regression("51059")
+def test_config_include_similar_name(tmp_path: pathlib.Path):
+    config_a = tmp_path / "a" / "config"
+    config_b = tmp_path / "b" / "config"
+
+    os.makedirs(config_a)
+    with open(config_a / "config.yaml", "w", encoding="utf-8") as fd:
```

Comment:
if you're using pathlib, you might as well stop using `os`? https://docs.python.org/3/library/pathlib.html#corresponding-tools, and `write_text`.